src/lib.rs

```diff
@@ -1953,6 +1953,15 @@ fn swap<T, U>((t, u): (T, U)) -> (U, T) {
 /// overhead. This is useful whenever memory is known to be in a zeroed state,
 /// such memory returned from some allocation routines.
 ///
+/// # Warning: Padding bytes
+///
+/// Note that, when an object is moved or copied, only the non-padding bytes of
```

Comment:
Rust does not have objects. We have allocations and typed or untyped copies. Untyped copies copy all bytes in the specified range, and currently the only way you can do this is `ptr::copy` and `ptr::copy_nonoverlapping`.

Typed copies are every other copy, which includes (some of these may be surprising to users) assignment, reads/writes through pointers, function arguments, and function return values.

Typed copies are not guaranteed to preserve bytes which are padding in all variants. That's probably what this comment means to identify.

Reply:
Yeah, "typed copy" is what I'm referring to. Do you know if that term is defined anywhere in the Reference or stdlib docs? A quick Google doesn't turn anything up.

Reply:
The docs for ptr::copy contain a probably-unhelpful mention of "untyped".

Reply:
Do you know of any internal docs/discussions that define the term? Or is this just one of those "compiler people use these terms and know what they mean when they talk to each other and they weren't expecting you to look inside the compiler" things?

Reply:
Ugh of course that doesn't work. 
@CAD97 @digama0 @JakobDegen @RalfJung do we have documentation for what a typed copy is?

Reply:
I don't know of any proper docs for this, no. That's clearly something we should fix. Any volunteers? :)

One question is where these docs would even go. The primary syntax for a typed copy is `a = b;`, so maybe this should be [here](https://doc.rust-lang.org/nightly/reference/expressions/operator-expr.html#assignment-expressions)?

Reply:
[s/object/value](https://github.com/google/zerocopy/pull/1660#discussion_r1761911299), but otherwise I'm gonna merge this and we can nitpick the wording later once docs have landed to Rust that we can cite.